dlt-connector/src/client/IotaClientSingleton.ts

```diff
@@ -0,0 +1,88 @@
+import { Client, ClientBuilder } from '@iota/client'
+import { MessageWrapper } from '@iota/client/lib/types'
+
+import CONFIG from '@/config'
+import { logger } from '@/server/logger'
+
+// Source: https://refactoring.guru/design-patterns/singleton/typescript/example
+// and ../federation/client/FederationClientFactory.ts
+/**
+ * A Singleton class defines the `getInstance` method that lets clients access
+ * the unique singleton instance.
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+class IotaClientSingleton {
+  // eslint-disable-next-line no-use-before-define
+  private static instance: IotaClientSingleton
+  private client: Client
```

Comment:
Ich verstehe hier nicht ganz, warum du instance nicht vom Typ `Client` hast, sondern das ganze doppelt schachtelst.
Dadurch könnten ggf. so Dinge wie `IotaClientSingleton.instance().instance().instance().instance().instance().instance()...` möglich sein - ich gebe zu ich habe mich hier nicht näher beschäftigt und ich kann mich irren oder es mag gar eine Designentscheidung sein, aber ggf erhöhst du hier unnötig die Komplexität. Außerdem sieht es fast so aus, als ob du mit `new IotaClientSingleton()` den Singleton-Mechanismus an sich aushebelst und damit das ganze Konstrukt zwar hinter einem Aufruf kapselst, aber nicht nur eine Instanz sicherstellst. Das Singleton stellt sicher, dass niemals new auf der Singleton-Klasse aufgerufen wird - auch nicht intern.
vgl: https://github.com/gradido/gradido/blob/master/backend/src/federation/client/FederationClientFactory.ts#L19

Reply:
Ich weiß das man bei Tests normalerweise alles mockt, aber mein Auftrag von Bernd war es Test Messages auf den Iota-Tangle zu bringen und wieder einzulesen. Da ich kein Interface habe, habe ich diesen Weg gewählt. 
Aber da es jetzt ein eigenständiges Modul ist, kann ich das auch in die main function tun.

Reply:
Du hast in meinem letzten PR geschrieben ich soll eine Singleton Klasse verwenden oder zumindest habe ich das so verstanden und das habe ich gemacht. 
Hier wird beschrieben wie man eine Standard Singleton Klasse in TypeScript definiert:
https://refactoring.guru/design-patterns/singleton/typescript/example
Ich habe mich nicht an deiner geposteten nicht instancierbaren Klasse orientiert, da diese in meinen Augen keinen Sinn macht so wie sie definiert ist. Man könnte bei Fall die Klasse auch einfach weg lassen und nur eine Globale Variable und Globale Funktionen benutzen. Oder man verwendet das tatsächliche Factory Design Pattern: https://refactoring.guru/design-patterns/factory-method/typescript/example

Reply:
Aber weißt du was, da Iota nun in einem eigenem Modul ist brauche ich eigentlich nur eine Globale Variable. Denn wenn die Initalisierung von Iota fehlschlägt, kann das Modul eh nicht arbeiten.